### PR TITLE
Enable python dispatcher for reinplacing pass

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #137097
* __->__ #137091

Arguably this should be put somewhere higher up in the stack?  Not sure.

Xref: https://fb.workplace.com/groups/6829516587176185/permalink/8042762615851570/

There is a repro but I need to fix more bugs before it can be checked in

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/fx_passes/reinplace.py [hunk 1]
```diff
@@ -708,6 +709,7 @@ def tensor_with_same_storage_already_reinplaced(arg):
 
 
 def reinplace_inplaceable_ops(graph: torch.fx.Graph) -> None:
-    canonicalize_view_scatter_ops(graph)
-    reinplace_inplaceable_ops_core(graph)
-    decompose_generalized_scatter(graph)
+    with enable_python_dispatcher():
```
Comment (Skylion007) [inline torch/_inductor/fx_passes/reinplace.py:712]: Can we just decorate the entire function reinplace_inplaceable_ops?

**Review by bdhirsh — APPROVED** [review]
test?

**Comment by ezyang** [comment]
Won't be landing this until after PTO

File: torch/_inductor/fx_passes/reinplace.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/fx_passes/reinplace.py:712]: In fact, I'm kind of thinking we should decorate the entire compiler

**Comment by ezyang** [comment]
@bdhirsh test is in the next PR

**Pull request closed without merging**